FHEM/14_SD_WS07.pm

```diff
@@ -206,12 +206,12 @@ SD_WS07_Parse($$)
 				$diffTemp = ($oldTemp - $temp);
 			}
 			$diffTemp = sprintf("%.1f", $diffTemp);				
-			Log3 $name, 4, "$iohash->{NAME}: $name old temp $oldTemp, age $timeSinceLastUpdate, new temp $temp, diff temp $diffTemp";
+			Log3 $name, 4, "$name: old temp $oldTemp, age $timeSinceLastUpdate, new temp $temp, diff temp $diffTemp";
 			my $maxDiffTemp = $timeSinceLastUpdate / 60 + $maxdeviation; 			# maxdeviation + 1.0 Kelvin/Minute
```

Comment:
Jetzt fehlt aber das IO-Device am Beginn der Log-Ausgabe. In $name steht "SD_WS07_TH_xx".

Reply:
Wieso sollen wir den Namen des IO Devices in einem logischen Modul ausgeben?

Reply:
Weil es zum einen sehr praktisch ist, wenn man sofort sieht, welcher Empfänger den Eintrag verursacht hat. Die meisten betreiben ja sicher mehr als einen Empfänger und manche Protokolle werden auch auf 2 Frequenzen verwendet. Zum anderen ist es üblich, das Device mit anzugeben.

Reply:
Hast Du mal ein Beispiel wo in der Logausgabe eines logischen Modules das Physische mit benannt wird?

Reply:
Ja, freilich auch mehrere:

2018.04.17 18:35:03 4: CUL_Parse: SCC_0 T521900A60032 -49
2018.04.07 15:25:43 3: CUL_0: CUL_WS_01 ERROR - Unknown temp value (temp=F2.3)
2018.04.25 10:55:38 4: sduino_USB0 Hideki_crc: rawdata=7512BACAEFBEEFA9 to short, count=7 data length=8
2018.04.25 11:09:26 3: sduino_USB0 IT: IT_FFFFFFF0 off->on
2018.04.25 11:10:04 3: sduino_USB0 IT: Code 1100 not supported by IT_FFFFFFF0.
2018.04.15 22:14:58 3: sduino: WS7053 ERROR - Ident not 1010 0000 - 01010000
2018.04.13 18:13:03 2: sduinoIP: CUL_TCM97001 Unknown device CUL_TCM97001_244, please define it

Das beste Beispiel, das diese Angaben sinnvoll sind, hatte ich nach dem letzten Update. Im Log tauchten des Öfteren Meldungen dieser Art auf:

2018.04.15 11:58:34 3: UNDEFINED Remotebutton send to define: DB1EF1_0_0
2018.04.15 12:02:20 3: UNDEFINED Remotebutton send to define: DB1EF1_0_0
2018.04.15 12:05:12 3: UNDEFINED Remotebutton send to define: DB1F11_0_0

Da hat man nicht die leiseste Ahnung, was diese Meldungen hervorruft. Ich habe dann die Log-Ausgabe erst mal bei mir auf die Schnelle geändert, so das sie jetzt so aussieht:

2018.04.15 12:17:50 3: sduino434 RSL_Parse UNDEFINED Remotebutton send to define: DB1EF1_0_0

Damit weiß ich sofort, bei welchem Empfänger ich welches Protokoll in die Blacklist nehmen muss.

Reply:
Es sieht halt irgendwie so aus, als ob das Log nicht vom logischen Modul kommt. Das verwirrt mich eher.
`2018.04.13 18:13:03 2: sduinoIP: CUL_TCM97001 Unknown device CUL_TCM97001_244, please define it`

Ich sehe ja dass es so ist, aber sonst steht doch an 1, stelle entweder das Device welches die Logausgabe verursacht hat oder halt das Modul. Ich hätte ja eher so etwas wie


`2018.04.13 18:13:03 2: CUL_TCM97001: Unknown device CUL_TCM97001_244, via (sduinoIP) please define it` oder meinetwegen

`2018.04.13 18:13:03 2: sduinoIP -> CUL_TCM97001: Unknown device CUL_TCM97001_244, please define it` 

gemacht.

Reply:
@sidey79 deinen Vorschlag kann man gern umsetzen.

die Variante, genannt von @elektron-bbs bzw. so wie ich sie erstmals drin habe, entspricht dem Schema wie es aus anderen Modulen kommt. Wollen wir ein "weiteres Schema" einbringen oder man verfährt eben so wie bei anderen Modulen es meist ist. Der Vorteil von "einem" einheitlichem Bild, das die Logausgabe im Logfile von FHEM einheitlicher erscheint.

Bsp: nochmal
>2018.04.27 13:06:47 3: nano_433Mhz: SD_WS07_TH_631 ERROR - Hum diff too large (old 46, new 11, diff 35)
2018.04.27 13:07:44 3: radino_433Mhz: SD_WS07_TH_631 ERROR - Hum diff too large (old 46, new 11, diff 35)

Erfahrungen zeigen, manche Module lassen den Doppelpunkt nur weg. Schema - Empfänger ---> Modul unsw...